util/math_reference.cpp

```diff
@@ -329,6 +339,106 @@ float clamp(const float a, const float b, const float c) {
   return clamp_t(a, b, c);
 }
 
+/* ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- DEGREES
+ *
+ */
+
+template <typename T>
+T degrees_t(T a) {
+  return a * (180.0L/M_PI);
+}
+
+float degrees(float a) {
+  return degrees_t(a);
+}
+
+double degrees(double a){
+  return degrees_t(a);
+}
+
+/* ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- RADIANS
+ *
+ */
+
+template <typename T>
+T radians_t(T a) {
+  return a * (M_PI/180.0L);
+}
+
+float radians(float a) {
+  return radians_t(a);
+}
+
+double radians(double a){
+  return radians_t(a);
+}
+/* ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- STEP
+ *
+ */
+
+template <typename T>
+T step_t(T a, T b) {
+  if (b < a)
+    return 0.0;
+  return 1.0;
+}
+
+float step(float a, float b) {
+  return step_t(a, b);
+}
+
+double step(double a, double b) {
+  return step_t(a, b);
+}
+
+/* ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- SMOOTHSTEP
+ *
+ */
+
+template <typename T>
+T smoothstep_t(T a, T b, T c) {
+  if (a >= b || std::isnan(a) || std::isnan(b) || std::isnan(c))
```

Comment:
Otherwise the reader will ask herself... "Wait... what is the result of `a >= b` just in case `a` or `b` is a NAN?..."
You can remove this frightening Halloween question by just making the NAN testing first everywhere.

Reply:
Changed here and in other places with isnan.